lib/finch/pool_manager.ex

```diff
@@ -1,19 +1,27 @@
 defmodule Finch.PoolManager do
   @moduledoc false
-  use Supervisor
+  use GenServer
 
-  alias Finch.PoolSup
-  alias Finch.PoolRegistry
+  def start_link(config) do
+    GenServer.start_link(__MODULE__, config, name: config.manager_name)
+  end
+
+  def init(config) do
+    Enum.each(config.pools, fn
+      {:default, _} -> :ok
+      {shp, _} -> do_start_pools(shp, config)
+      _ -> :ok
+    end)
 
-  def start_link(init_args) do
-    Supervisor.start_link(__MODULE__, init_args, name: __MODULE__)
+    {:ok, config}
   end
 
-  def get_pool(scheme, host, port) do
+  def get_pool(registry_name, scheme, host, port) do
     key = {scheme, host, port}
-    case lookup_pool(key) do
+
+    case lookup_pool(registry_name, key) do
       :none ->
-        case start_pool(key) do
+        case start_pools(registry_name, key) do
```

Comment:
`start_pools` should no longer return `{:error, _}`, correct? Because we have removed the race. So maybe we can remove the case?